app/src/main/kotlin/cz/covid19cz/erouska/db/SharedPrefsRepository.kt

```diff
@@ -77,7 +78,13 @@ class SharedPrefsRepository(c : Context) {
 
     fun getAppPaused() = prefs.getBoolean(APP_PAUSED, false)
 
+    fun getDbEncryptedKey(): String? = prefs.getString(DB_ENCRYPTED_KEY, null)
+
+    fun setDbEncryptedKey(key: String) {
+        prefs.edit().putString(DB_ENCRYPTED_KEY, key).apply()
```

Comment:
Why storing key into SharedPreferences? I thought it will be stored in the key store.

Reply:
It works like this: you create a key K1 which is used to encrypt the database. Then you encrypt K1 using the keystore and store it encrypted within shared prefs. When you need to work with DB, you take it from shared prefs, decrypt using keystore and use K1 to encrypt/decrypt DB.

Reply:
The key still lives for some time in RAM, but it is not stored on disk permanently, which provides added security.

Reply:
But why can't you store it in key store? Key store is just for encrypting/decrypting stuff but not for storing keys?

Reply:
If I understand it correctly, the key never leaves the keystore. Unencrypted data goes in, encrypted goes out and vice versa. So you can't use the keystore for directly encrypting the DB. But maybe I'm wrong :)